packages/consoledot-api/src/mutations/useDeleteConsumerMutation.ts

```diff
@@ -0,0 +1,43 @@
+import { isServiceApiError } from "@rhoas/kafka-management-sdk";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { kafkaQueries } from "../queryKeys";
+import { useApi } from "../useApi";
+
+export function useDeleteConsumerMutation() {
+  const { consumerGroups } = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    async function kafkaCreateInstance(props: {
+      instanceId: string;
+      adminUrl: string;
+      name: string;
+      onSuccess: () => void;
+      onError: (code: string, message: string) => void;
+    }) {
+      const { instanceId, adminUrl, name, onSuccess, onError } = props;
+      const api = consumerGroups(adminUrl);
+      try {
+        await api.deleteConsumerGroupById(name);
+        onSuccess();
+      } catch (error) {
+        if (isServiceApiError(error)) {
+          const message = error?.message;
+          const { code } = error?.response?.data || {};
+
+          onError(code || "?", message);
+        }
+      }
+    },
+    {
+      onSuccess: (_, { adminUrl, instanceId }) => {
+        void queryClient.invalidateQueries([
+          kafkaQueries.instance._root({
+            id: instanceId,
+            adminUrl,
+          }),
+        ]);
```

Comment:
There is really no need to invalidate the whole cache for this. We could invalidate the consumer groups related one tho.